server/datastore/mysql/scripts_test.go

```diff
@@ -800,3 +801,131 @@ func testUnlockHostViaScript(t *testing.T, ds *Datastore) {
 	require.False(t, status.IsPendingUnlock())
 	require.False(t, status.IsLocked())
 }
+
+func testLockUnlockViaScripts(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+	user := test.NewUser(t, ds, "Bob", "bob@example.com", true)
+
+	checkState := func(t *testing.T, status *fleet.HostLockWipeStatus, unlocked, locked, wiped, pendingUnlock, pendingLock, pendingWipe bool) {
+		require.Equal(t, unlocked, status.IsUnlocked())
+		require.Equal(t, locked, status.IsLocked())
+		require.Equal(t, wiped, status.IsWiped())
+		require.Equal(t, pendingLock, status.IsPendingLock())
+		require.Equal(t, pendingUnlock, status.IsPendingUnlock())
+		require.Equal(t, pendingWipe, status.IsPendingWipe())
+	}
+
+	for i, platform := range []string{"windows", "linux"} {
+		hostID := uint(i + 1)
+
+		t.Run(platform, func(t *testing.T) {
+			status, err := ds.GetHostLockWipeStatus(ctx, hostID, "windows")
```

Comment:
nit: would it be worth using `platform` as the final argument when calling this method? Would be clearer IMO.

Reply:
oh good point, I'll fix it in my upcoming PR, will merge this one in the meantime to keep things simpler.